### PR TITLE
address_module

### PR DESCRIPTION
**Pull request by nikpawar89**

### CODE REVIEW

File: fineract-provider/src/main/java/org/apache/fineract/infrastructure/configuration/api/GlobalConfigurationApiResource.java [hunk 1]
```diff
@@ -47,73 +47,74 @@
 @Scope("singleton")
 public class GlobalConfigurationApiResource {
```
Comment (nazeer1100126) [inline fineract-provider/src/main/java/org/apache/fineract/infrastructure/configuration/api/GlobalConfigurationApiResource.java:48]: Since there is no change in this file, this file should not be part of this PR

File: fineract-provider/src/main/resources/sql/migrations/core_db/V312__address_module_tables_metadat.sql [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+
+-- code inserts
+INSERT INTO `m_code` (`code_name`, `is_system_defined`) VALUES ('STATE',1);
+INSERT INTO `m_code` (`code_name`, `is_system_defined`) VALUES ('COUNTRY',1);
+INSERT INTO `m_code` (`code_name`, `is_system_defined`) VALUES ('ADDRESS_TYPE',1);
+
+-- inserts for STATE
+INSERT INTO `m_code_value` (`code_id`, `code_value`,`order_position`, `is_active`) 
+VALUES(
+ (select id from m_code where code_name = 'STATE'), 'Andhra Pradesh', 0,1);
```
Comment (nazeer1100126) [inline fineract-provider/src/main/resources/sql/migrations/core_db/V312__address_module_tables_metadat.sql:29]: I don't think we need to add states of India as part of SQL migration. This is admin job and he will be adding countries and states from the UI.

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/address/api/AddressConfigurationApiResources.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+import org.apache.fineract.portfolio.address.service.AddressReadPlatformServiceImpl;
+import org.apache.fineract.portfolio.address.service.FieldConfigurationReadPlatformService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Scope;
+import org.springframework.stereotype.Component;
+
+@Path("/clients/addresses")
+@Component
+@Scope("singleton")
+public class AddressConfigurationApiResources {
```
Comment (nazeer1100126) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/address/api/AddressConfigurationApiResources.java:51]: I believe this template api should be part of ClientAddressApiResources. And this class can be removed

File: fineract-provider/src/main/java/org/apache/fineract/portfolio/address/domain/Address.java [hunk 4]
```diff
@@ -0,0 +1,348 @@
...
+
+	/*@OneToMany(mappedBy = "address", cascade = CascadeType.ALL)
+	private List<ClientAddress> clientaddress = new ArrayList<>();*/
+	
+	@OneToMany(mappedBy = "address", cascade = CascadeType.ALL)
+	private Set<ClientAddress> clientaddress;
+
+	private String street;
+
+	private String address_line_1;
```
Comment (nazeer1100126) [inline fineract-provider/src/main/java/org/apache/fineract/portfolio/address/domain/Address.java:50]: Give proper names to all variables in entity classes. You can refer other entity classes

**Comment by Nayan** [comment]
@nikpawar89 please add the API docs :)
